File: confusing.py
num = int(input("enter number"))
flag = False
digit = 0
while num > 0:
    digit = int(num % 10)
    if digit != 0 and digit !=1 and digit !=6 and digit !=8 and digit !=9:
        flag = True
        break
    num=int(num/10)
if flag==False:
    print("confusing number")
else:
    print("not a confusing number")

File: confusing.c
#include <stdio.h>
main()
{
    int num = 0, digit = 0, flag = 0;
    printf("enter any number");
    scanf("%d", &num); // 689
    while (num > 0)
    {
        digit = num % 10; // fetch the digit 9 8 6

        if (digit != 0 && digit != 1 && digit != 6 && digit != 8 && digit != 9)
        {
            flag = 1;
            break; // forcefully terminate
        }
        num = num / 10; // 689 => 68 =>6 =>0
    }

    if (flag == 0)
    {
        printf("confusing number");
    }
    else
    {
        printf("not a confusing number");
    }
}